File: Backend3/src/main/java/edu/cit/backend3/dto/MemberDto.java
package edu.cit.backend3.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MemberDto {
    private Long id;
    private String username;
    private String email;
    private String firstName;
    private String lastName;
    private String role;
    private List<String> roles;
    private String status;
    private String createdAt;
    private String phoneNumber;
    private String studentNumber;
    private String city;
    private String province;
    private String address;
    private String bio;
}

File: MobileDev/app/src/main/java/edu/cit/cituforumsmobile/api/dto/MemberDto.kt
package edu.cit.cituforumsmobile.api.dto

data class MemberDto(
    val id: Long,
    val username: String, // Changed from name
    val email: String,
    val firstName: String?, // Use nullable types for fields that might be null
    val lastName: String?,
    val role: String?,
    val roles: List<String>?,
    val status: String?,
    val createdAt: String?,
    val phoneNumber: String?,
    val studentNumber: String?,
    val city: String?,
    val province: String?,
    val address: String?,
    val bio: String?
)